### PR TITLE
feat: CreateMessage command

### PR DESCRIPTION
**Pull request by jerotire**
## Description

Created the CreateMessage command; allows for the creation of Message and MessageContent entities and attaching to Conversations.

Some QOL improvements too, for ensuring persists and removes cascades to the MessageContent table.

Related issue: [VOL-4572](https://dvsa.atlassian.net/browse/VOL-4572)

## Before submitting (or marking as "ready for review")

- [x] Does the pull request title follow the [conventional commit](https://www.conventionalcommits.org/en/v1.0.0/) specification?
- [x] Have you performed a self-review of the code
- [x] Have you have added tests that prove the fix or feature is effective and working
- [ ] Did you make sure to update any documentation relating to this change?

### CODE REVIEW

File: module/Api/src/Domain/CommandHandler/Messaging/Message/Create.php [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+    protected $toggleConfig = [FeatureToggle::MESSAGING];
+    protected $extraRepos = [
+        ConversationRepo::class,
+        MessageRepo::class,
+        MessageContentRepo::class,
+    ];
+
+    public function handleCommand(CommandInterface $command)
+    {
+        assert($command instanceof CreateMessageCommand);
```
Comment (wadedvsa) [inline module/Api/src/Domain/CommandHandler/Messaging/Message/Create.php:37]: S: Remove assert. Bit of a discussion on one of Saul's tickets a few days ago. You can turn them off in PHP.ini so can't be relied on (PHP docs say only use for debugging/testing).

File: module/Api/src/Domain/Repository/RepositoryFactory.php [hunk 2]
```diff
@@ -9,9 +9,11 @@ class RepositoryFactory implements FactoryInterface
 {
     public function __invoke(ContainerInterface $container, $requestedName, array $options = null)
     {
-        $class = __NAMESPACE__ . '\\' . $requestedName;
+        if (!class_exists($requestedName)) {
```
[the remark this answers is not in the document]
Comment (wadedvsa) [reply module/Api/src/Domain/Repository/RepositoryFactory.php:12]: Is this a safe change considering it's a base class logic change?

**Review by wadedvsa — DISMISSED** [review]
Couple of comments, a should around assertions. Hard to tell from this but does this cover attachments to messages?

**Comment by hobbyhacker0** [comment]
> Is this a safe change considering it's a base class logic change?

Also wondering if this is safe, probably OK but wondering if there are quirks somewhere

**Review by hobbyhacker0 — APPROVED** [review]

**Pull request merged**